data_tools/featurizer.py

```diff
@@ -0,0 +1,40 @@
+import streamlit as st
+import pandas as pd
+import matplotlib.pyplot as plt
+
+
+def get_page_count(first_page, last_page):
+    return [
+        int(l) - int(f) if l and f is not None else None
+        for f, l in zip(first_page, last_page)
+    ]
+
+
+le = {}
```

Comment:
Instead of using a global for `le`, could `encode_categorical` return `le` and `decode_categorical` could accept it as an argument?